### PR TITLE
[SPARK-12692][BUILD][STREAMING] Scala style: Fix the style violation (Space before "," or ":")

### PR DESCRIPTION
**Pull request by sarutak**
Fix the style violation (space before , and :).
This PR is a followup for #10643.

### CODE REVIEW

File: external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/Logging.scala [hunk 1]
```diff
@@ -26,20 +26,20 @@ import org.slf4j.{Logger, LoggerFactory}
 private[sink] trait Logging {
   // Make the log field transient so that objects with Logging can
   // be serialized and used on another machine
-  @transient private var log_ : Logger = null
```
Comment (sarutak) [inline external/flume-sink/src/main/scala/org/apache/spark/streaming/flume/sink/Logging.scala]: We can't write like `log_: Logger`. So, can we rename those vars to which starts with `_`?

**Comment by rxin** [comment]
LGTM - the mima check is related to the logging trait change.

**Comment by sarutak** [comment]
The reason of the MiMa failure is Maven Central doesn't provide `mvttv3;1.0.1` while `streaming-mqtt_2.10;1.6.0` depends on mqttv3;1.0.1 right?

```
[info] 	[SUCCESSFUL ] org.apache.spark#spark-streaming-mqtt_2.10;1.6.0!spark-streaming-mqtt_2.10.jar (21ms)
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	::          UNRESOLVED DEPENDENCIES         ::
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
[warn] 	:: org.eclipse.paho#org.eclipse.paho.client.mqttv3;1.0.1: not found
[warn] 	::::::::::::::::::::::::::::::::::::::::::::::
```

**Comment by sarutak** [comment]
#10683 and #10678 may hit the same MiMa problem

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by JoshRosen** [comment]
Jenkins, retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
retest this please.

**Comment by sarutak** [comment]
I think the PySpark unit test failures above are related to #10704 and already fixed.

**Comment by sarutak** [comment]
retest this please.

**Comment by rxin** [comment]
Thanks - I'm going to merge it.

**Comment by rxin** [comment]
@sarutak I actually reverted this change after talking more with Michael and looking at the diffs. The problem is that for colon, in some cases it is really useful to have a space before colon. In particular:

1. In many cases for type bounds, ti is good to have a space before the colon so it is more clear.
2. For type descriptions (e.g. variable : Int), it is also often useful to have a space.

I don't know whether it is feasible to actually implement the rules for the general colon, but exclude type description and type bounds. If yes, we should add this enforcement. Otherwise I'd prefer not to have it at all. 

For comma, we should always have it (at least I haven't seen a case in which we don't want it yet).

**Comment by sarutak** [comment]
```
1. In many cases for type bounds, ti is good to have a space before the colon so it is more clear.
2. For type descriptions (e.g. variable : Int), it is also often useful to have a space.
```

It exactly make sense. O.K, let's check style violation only for comma.
Only the change for SQL is reverted for now so would you like me to  revert the rest of the changes?

**Comment by rxin** [comment]
The rest are ok since they are small. I reverted the sql one because it was large.

**Pull request closed without merging**